src/en/clowncorps/src/eu/kanade/tachiyomi/extension/en/clowncorps/ClownCorps.kt

```diff
@@ -0,0 +1,241 @@
+        url = "/comic"
+    }
+
+    override fun fetchPopularManga(page: Int): Observable<MangasPage> =
+        Observable.just(MangasPage(listOf(getManga()), hasNextPage = false))
+
+    override fun fetchSearchManga(page: Int, query: String, filters: FilterList) =
+        fetchPopularManga(page)
+
+    override fun fetchMangaDetails(manga: SManga): Observable<SManga> =
+        Observable.just(getManga())
+
+    @Serializable
+    data class SerializableChapter(val fullLink: String, val name: String, val dateUpload: Long) {
+        override fun hashCode(): Int {
+            return name.hashCode()
+        }
+
+        override fun equals(other: Any?): Boolean {
+            if (this === other) return true
+            if (javaClass != other?.javaClass) return false
+
+            other as SerializableChapter
+
+            return name == other.name
+        }
+    }
+
+    override fun chapterListParse(response: Response): List<SChapter> {
+        // The total number of webpages with chapters on them
+        val currentPageIndicator = response.asJsoup().select("#paginav li.paginav-pages").text()
+        val totalWebpageCount = currentPageIndicator.split(" ").last().toInt()
+
+        val allChapters = getChaptersFromCache().toMutableSet()
+        // Fetch all the chapters from the website until we reached where the cache left off
+        for (webpageIndex in 1..totalWebpageCount) {
+            val anyChaptersWereAdded = allChapters.addAll(fetchWebpageChapters(webpageIndex))
+            if (!anyChaptersWereAdded) break // No new chapters were added from this webpage, so we're done
+        }
+
+        // Save the chapters to cache
+        val fullJsonString = Json.encodeToString(allChapters)
+        preferences.edit().putString(CACHE_KEY_CHAPTERS, fullJsonString).apply()
+
+        // Convert the serializable chapters to SChapters
+        return allChapters.map { chapter ->
+            SChapter.create().apply {
+                setUrlWithoutDomain(chapter.fullLink)
+                name = chapter.name
+                date_upload = chapter.dateUpload
+            }
+        }
+    }
+
+    private fun getChaptersFromCache(): Set<SerializableChapter> {
+        val cachedChaps = preferences.getString(CACHE_KEY_CHAPTERS, null)
+        if (cachedChaps != null) return Json.decodeFromString(cachedChaps)
+        return emptySet()
+    }
+
+    private fun fetchWebpageChapters(webpageIndex: Int): List<SerializableChapter> {
+        val url = "$baseUrl/comic/page/$webpageIndex/"
+        val resp = client.newCall(GET(url, headers)).execute()
+        val comics = resp.asJsoup().select(".comic")
+        return comics.mapNotNull {
+            val link = it.selectFirst(".post-title a")?.attr("href")
+            val title = it.selectFirst(".post-title a")?.text()
+            val postDate = it.selectFirst(".post-date")?.text()
+            val postTime = it.selectFirst(".post-time")?.text()
```

Comment:
Any reason they could be missing? If not, use `!!`.

Reply:
Probably not, but in case they do happen to be missing, should it really crash at runtime instead of handling it properly?

Reply:
It would cause the app to show an error instead, so you could identify the change on the website. Ignoring the error is not "handling it properly".

Reply:
Fair enough :+1:

Reply:
Done in 6cb092ddbb898d6163edb84230985b7ca40c5b6b